x-pack/plugin/watcher/src/main/java/org/elasticsearch/xpack/watcher/WatcherLifeCycleService.java

```diff
@@ -166,7 +166,9 @@ public void clusterChanged(ClusterChangedEvent event) {
             if (watcherService.validate(event.state())) {
                 previousShardRoutings.set(localAffectedShardRoutings);
                 if (state.get() == WatcherState.STARTED) {
-                    watcherService.reload(event.state(), "new local watcher shard allocation ids");
+                    watcherService.reload(event.state(), "new local watcher shard allocation ids", (exception) -> {
+                        clearAllocationIds(); // will cause reload again
```

Comment:
Do we need to also set the state to WatcherState.STARTING here? It's been a couple of years since i've been in here, so I'm not sure what bad things might happen if the state is STARTED but watcher is actually unavailable.

Reply:
@masseyke I understand the point but I don't think so. If the state changed to `STARTING` here, the watcher service would no longer be reloaded, which is the problematic state that was also reported in #44981. Since there are only four states, `STARTED` (but it's rather DEGRADED actually) would be affordable, in my opinion.

Reply:
OK that makes sense. And it's not going to be worse than the current situation (having it in STARTED state, but not doing anything).